pkg/annotations/dynamic-network-status.go

```diff
@@ -64,6 +66,31 @@ func PodDynamicNetworkStatus(currentPod *corev1.Pod) ([]nettypes.NetworkStatus,
 	return currentIfaceStatus, nil
 }
 
+// SetNetworkStatus updates the Pod status
+func SetNetworkStatus(pod *corev1.Pod, statuses []v1.NetworkStatus) error {
```

Comment:
hm, maybe I'm missing something, but why are you implementing this function instead of re-using the one provided in the [network attachment definition client](https://github.com/k8snetworkplumbingwg/network-attachment-definition-client/blob/93ac89453ffd471cc943eb255799f627777eb1a8/pkg/utils/net-attach-def.go#L49) ? 

We're already importing those utils. 

If there's a reason for that, please also refer it in the commit message.

Reply:
This function in the nad-client will keep the pod object passed in parameter intact. It fetches the pod via the name from the Kubernetes API, updates its annotation and updates it via the Kubenretes API.
https://github.com/k8snetworkplumbingwg/network-attachment-definition-client/blob/v1.4.0/pkg/utils/net-attach-def.go#L87

The function I created is different. It updates the annotation of the pod object passed as parameter without updating it in Kubernetes. 
So, when the add happens, the pod object is modified with the new annotations (not yet updated in Kubernetes). The del will then happen with the pod object modified with the added annotation (here is the difference with the previous behavior), and then the del will modify the annotations and then the Kubernetes API is called to update the pod itself.